js/src/setup-mc/setup-stepper/setup-free-listings/shipping-time/shipping-time-setup/countries-time-input/edit-time-button/edit-time-modal/index.js

```diff
@@ -80,7 +81,9 @@ const EditTimeModal = ( props ) => {
 
 				return (
 					<AppModal
-						className="gla-edit-time-modal"
+						overflowY="visible"
+						shouldCloseOnEsc={ ! dropdownVisible }
+						shouldCloseOnClickOutside={ ! dropdownVisible }
```

Comment:
💅  What about just a property like `dropdownVisible ={ dropdownVisible }` ??

My concerns with the current implementation are that we need to set up 3 props in order to make the modal support the dropdown.

- Set overflowY="visible" (is only "visible" if there is a dropdown)
- Set shouldCloseOnEsc={ ! dropdownVisible } (related to dropdown)
- Set shouldCloseOnClickOutside={ ! dropdownVisible } (Related to dropdown)

Hence, having that `withDropdown={ dropdownVisible }` or creating a wrapper component `AppWithDropdown` will be better in terms of code reusability. 

```
const AppModal = ( {
	className,
	dropdownVisible,
	buttons = [],
	children,
	...rest
} ) => {
	const modalClassName = classnames(
		'app-modal',
		dropdownVisible && `app-modal__styled--overflow-y-visible`,
		className
	);

    <Modal 
          className={ modalClassName } 
          shouldCloseOnEsc={ ! dropdownVisible }
	      shouldCloseOnClickOutside={ ! dropdownVisible }>
```

Reply:
> What about just a property like `dropdownVisible ={ dropdownVisible }` ??

Usually, the more basic the component, the less interconnectivity between its props internally, and these interconnections are more appropriate for higher-level (shared) components. 

The AppModal is a basic component for generic uses. Using a `dropdownVisible` prop to control the other three props will reduce flexibility. For example, a modal that doesn't allow to be closed by clicking on outside area or pressing ESC, regardless of whether the dropdown is displayed or not.

> or creating a wrapper component `AppWithDropdown` will be better in terms of code reusability.

Adding another layer of wrapper components might help the reusability, but I lean toward reducing the code conflicts to the in-progress adjustments of shipping modals.

Reply:
I disagree that the current implementation reduces complexity. Adding  `shouldCloseOnEsc` &   `shouldCloseOnClickOutside` in the AppModal component based on the dropdown property would reduce the complexity for the implementation since it would needed to send 1 parameter instead of 3. Besides the fact that adding that those parameters seems for me potentially forgettable. 

> The AppModal is a basic component for generic uses. Using a dropdownVisible prop to control the other three props will reduce flexibility. For example, a modal that doesn't allow to be closed by clicking on outside area or pressing ESC, regardless of whether the dropdown is displayed or not.

Adding  `shouldCloseOnEsc` &   `shouldCloseOnClickOutside` in the AppModal component based on the dropdown property is not reducing flexibility since we can use also the ...rest feature and add directly `shouldCloseOnEsc` &   `shouldCloseOnClickOutside` as before.

Saying that and since its not a blocking issue, I rely on your criteria (or further PRs) to consider that or not

Reply:
> I disagree that the current implementation reduces complexity.

I meant that the flexibility would be reduced if using a prop to control the other three props internally and make interconnections on them. I supposed I didn't mention which complexity is lower. If I were to evaluate the complexity, I don't think the difference is significant between both.

> Adding `shouldCloseOnEsc` & `shouldCloseOnClickOutside` in the AppModal component based on the dropdown property would reduce the complexity for the implementation since it would needed to send 1 parameter instead of 3. Besides the fact that adding that those parameters seems for me potentially forgettable.

> Adding `shouldCloseOnEsc` & `shouldCloseOnClickOutside` in the AppModal component based on the dropdown property is not reducing flexibility since we can use also the ...rest feature and add directly `shouldCloseOnEsc` & `shouldCloseOnClickOutside` as before.

Surely this could make it less needed props for the cases in this PR, but I meant it doesn't make the opposite use case easier. If a floating block is showing and inside a modal, but the scenario doesn't need to prevent that modal from being closed by pressing ESC or by clicking on outside, then it still needs to specify the AppModal like:
```js
<AppModal
  dropdownVisible={ dropdownVisible }
  shouldCloseOnEsc={ true }
  shouldCloseOnClickOutside={ true }
>
```

The need of adding props like `shouldCloseOnEsc` and `shouldCloseOnClickOutside` is still the same as "potentially forgettable". Preserving the independence of their logic is a way to retain flexibility for a basic level component.

---

But considering the current instances are the same case, I would say it's still fine to use a single prop approach you suggested. However, I will have two more concerns with that approach:
1. The prop name `dropdownVisible` might not be ideal. Suppose the AppModal should not need to be too specific and literal, such as **dropdown**. It would be more appropriate to use a name that describes the intent, otherwise coupling in the naming is higher.
2. Since the given single prop is a dynamic state, and its rendering is not updated in the same tick as the updating of Modal `className`, this causes visible flickering on slower computer hardware.
   ![Kapture 2022-04-27 at 12 19 17](https://user-images.githubusercontent.com/17420811/165440585-28b2ffa8-1203-4cd4-9e7e-13a4cc88eb35.gif)

I'd like to know if you have a better idea to avoid these concerns.